Plugins/NativeEngine/Source/VertexBuffer.cpp

```diff
@@ -75,7 +75,22 @@ namespace Babylon
         else
         {
             // Buffer hasn't been finalized yet, all that's necessary is to swap out the bytes.
-            m_bytes = {bytes.data(), bytes.data() + bytes.size()};
+            if (m_bytes && !(*m_bytes).empty())
+            {
+                // update a portion of the vertex buffer bytes
+                memcpy((*m_bytes).data() + byteOffset, bytes.data(), bytes.size());
```

Comment:
This seems dangerous. What happens if someone passes an out-of-bounds byteOffset?

Reply:
True. Adding a check.